src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -137,14 +137,14 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                         return (int)ReturnCode.ArgumentError;
                     }
 
-                    collectRundownEvents = selectedProfile.Rundown;
+                    getRundownKeyword = selectedProfile.GetRundownKeyword;
 
                     Profile.MergeProfileAndProviders(selectedProfile, providerCollection, enabledBy);
                 }
 
                 if (rundown.HasValue)
                 {
-                    collectRundownEvents = rundown.Value;
+                    getRundownKeyword = _ => rundown.HasValue ? EventPipeSessionConfiguration.DefaultRundownKeyword : 0;
```

Comment:
If the `getRundownKeyword(...)` from the profile specified a rundown keyword with more bits than `DefaultRundownKeyword`, this would have cleared it.

Reply:
Good catch. I wasn't too focused on complete correctness with this prototype, but just wanted to demonstrate my idea. Depending on if this is adopted and who does the work, we should integrate a fix for this.